### PR TITLE
Change blockingTaskExecutor from Executor to ScheduledExecutorS…

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
It will be good if a user can schedule a task using `blockingTaskExecutor`.

Modification:
- Change `blockingTaskExecutor` from `Executor` to `ScheduledExecutorService`.

Result:
- You can now schedule a task using `blockingTaskExecutor`.
- (Breaking) `ServerBuilder.blockingTaskExecutor(Executor executor, ...)` is now takes `ScheduledExecutorService`.

### CODE REVIEW

**Comment by minwoox** [comment]
@renaudb You might want to see this because you are using the customized `ThreadPoolExecutor`.

**Comment by trustin** [comment]
@renaudb I think you can still achieve the same effect by taking account of the number of pending tasks using an `AtomicInteger` and rejecting the request before calling `ExecutorService.execute()`. However, if you think the inability to specify a `SynchronousQueue` as a task queue is a problem, as a follow-up PR, I guess we can provide a `ScheduledExecutorService` implementation which is a hybrid of `ThreadPoolExecutor` and `ScheduledThreadPoolExecutor`:

```java
public class HybridExecutorService implements ScheduledExecutorService {

    ThreadPoolExecutor executor = ...;
    ScheduledThreadPoolExecutor scheduledExecutor = ...;

    public void execute(Runnable task) { 
        executor.execute(task);
    }

    public ScheduledFuture<?> schedule(Runnable task, ...) {
        scheduledExecutor.schedule(() -> execute(task), ...);
        ... some future delegation ...
    }
}
```

Thoughts?

**Comment by ikhoon** [comment]
nit: Don't we need to update the `blockingTaskExecutor` documentation in `Production checklist`?
https://github.com/line/armeria/blob/f03b400cff136a710dbdbeb75e0e4f741387d296/site/src/sphinx/advanced-production-checklist.rst#L22-L25

**Comment by renaudb** [comment]
No, this is fine, we might just assign a specific context aware ExecutorService just to Solr and build our own API on top to make it look async.

**Comment by minwoox** [comment]
> nit: Don't we need to update the blockingTaskExecutor documentation in Production checklist?

Thanks a lot for finding this. 😄

**Comment by minwoox** [comment]
@ikhoon Fix. PTAL 🙇

File: site/src/sphinx/advanced-production-checklist.rst [hunk 1]
```diff
@@ -21,15 +21,15 @@ You may want to consider the following options before putting your Armeria appli
...
+  synchronous service implementation. The default is a simple ``ScheduledThreadPoolExecutor`` with maximum
+  200 threads, provided by :api:`CommonPools`.
 
   .. code-block:: java
 
       import com.linecorp.armeria.server.ServerBuilder;
 
       ServerBuilder sb = Server.builder();
-      sb.blockingTaskExecutor(myBoundedExecutor);
+      sb.blockingTaskExecutor(my600ThreadsExecutor);
```
Comment (ikhoon) [inline site/src/sphinx/advanced-production-checklist.rst]: nit: I'm not sure this suggestion is good. 😅 `myScheduledExecutorService`?

**Comment by ikhoon** [comment]
@minwoox Thanks for listening to my opinion 👍 Cool Beans!

File: site/src/sphinx/advanced-production-checklist.rst (hunk 1 above)
Comment (minwoox) [reply site/src/sphinx/advanced-production-checklist.rst]: The queue of default scheduled pool executor is always unbounded. So I used your first suggestion. 😉

**Comment by trustin** [comment]
Nice job, @minwoox !

**Comment by minwoox** [comment]
Thanks for reviewing! :wink:

**Pull request merged**